Sources/Mantis/Helpers/TransformStack.swift

```diff
@@ -0,0 +1,50 @@
+import UIKit
+
+
+class TransformStack: NSObject {
+    
+    static var shared: TransformStack = TransformStack()
+    
+    weak var transformDelegate : TransformDelegate?
+    
+    private var transformAdjustmentsStack: [TransformRecord] = []
+    
+    var top: Int = 0
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private override init() {
+        super.init()
+        
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.undoStatusChanged(notification:)),
+            name: .NSUndoManagerCheckpoint,
+            object: nil)
+    }
+    
+    func pushTransformRecord(_ record: TransformRecord) {
+        if transformAdjustmentsStack.count > top {
+            transformAdjustmentsStack.remove(at: top)
+        }
+        transformAdjustmentsStack.insert(record, at: top)
+        top += 1
+    }
+
+    func popTransformStack() {
+        top -= 1
+    }
```

Comment:
I think here we need to handle the case of `top < 0` if popTransformStack is repeatedly called.
That would be great if you could add some unit tests for the new code you added.

Reply:
added bounds check

Reply:
Added unit test for TransformStack.